src/components/general/SideSheet/Modal/index.tsx

```diff
@@ -0,0 +1,103 @@
+import React, { ReactNode, FC, useCallback, useMemo, useState, useEffect } from 'react';
+import styles from './styles.less';
+import classNames from 'classnames';
+import {
+    IconButton,
+    CloseIcon,
+    useElevationClassName,
+    useOverlayContainer,
+} from '@equinor/fusion-components';
+import { useComponentDisplayClassNames } from '@equinor/fusion';
+import { createPortal } from 'react-dom';
+
+type ModalSideSheetProps = {
+    children: ReactNode;
+    header?: string;
+    show?: boolean;
+    scrollable?: boolean;
+    onClose?: () => void;
+    headerIcons?: ReactNode[];
+};
+
+const ModalSideSheet: FC<ModalSideSheetProps> = ({
+    children,
+    header,
+    show,
+    scrollable,
+    onClose,
+    headerIcons,
+}) => {
+    const [isShowing, setIsShowing] = useState(false);
+
+    useEffect(() => {
+        if (!isShowing && show) {
+            setIsShowing(true);
+        }
+    }, [show]);
+
+    const close = useCallback(() => {
+        setIsShowing(false);
+    }, []);
+
+    const modalSideSheetClassNames = classNames(
+        styles.modalSideSheet,
+        useElevationClassName(16),
+        useComponentDisplayClassNames(styles)
+    );
+
+    const containerClassName = classNames(styles.container, {
+        [styles.show]: isShowing,
+    });
+    const content = useMemo(() => {
+        if (!show) {
+            return null;
+        }
+
+        if (scrollable) {
```

Comment:
Would this work if it was "scrollable" all the time? Could simplify the code and usage of the component quite a bit